ci/infra/testrunner/skuba/skuba.py

```diff
@@ -28,48 +28,42 @@ def _verify_bootstrap_dependency(self):
     @staticmethod
     @step
     def build(conf):
-        """Buids skuba from source"""
+        """Builds skuba from source"""
         utils = Utils(conf)
-        utils.runshellcommand("rm -fr go")
-        utils.runshellcommand("mkdir -p go/src/github.com/SUSE")
-        utils.runshellcommand("cp -a {} go/src/github.com/SUSE/".format(conf.skuba.srcpath))
+        go_dir = os.path.join(conf.workspace, "go")
+        src_dir = os.path.join(go_dir, "src/github.com/SUSE")
+
+        # Make a fresh copy of the source code
+        utils.cleanup_file(go_dir)
+        os.makedirs(src_dir, exist_ok=True)
+        shutil.copytree(conf.skuba.srcpath, os.path.join(src_dir, os.path.basename(conf.skuba.srcpath)))
+
         utils.gorun("go version")
         utils.gorun("make")
 
     @staticmethod
     @step
     def cleanup(conf):
         """Cleanup skuba working environment"""
-        utils = Utils(conf)
-        # TODO: check why (and if) the following two commands are needed
-        cmd = 'mkdir -p {}/logs'.format(conf.workspace)
-        utils.runshellcommand(cmd)
 
+        # TODO: check why (and if) the following command is needed
```

Comment:
nit: I prefer github issues over TODO's in code

Reply:
also. not sure what files this is chmod'ing BUT.... some files in skuba need to have restricted `600` permissions or skuba will reject them.

Reply:
i can also not explain this. the chmod should ideally only be files that are not tracked in the git tree, otherwise you will always have a git diff when you run testrunner within a clone. this is basically messing up the git clone and forces you to use testrunner only in a separate git clone, not to mess up the one you're mainly developing on. 

> nit: I prefer github issues over TODO's in code

:+1: an issue would be good to figure this out. when this is fixed i believe you could use testrunner also in your main git clone that you develop on, which makes things much easier

Reply:
The `chmod` is applied to files in the `workspace` which is a working directory and should not be under git. 

Test runner can be used with the project clone from git.